### PR TITLE
UI changes for unverified users

### LINKED ISSUE #128: Unverified account view (screen 201)
from #51 
  
  * [x] Don't block entire UX for unverified users.  Provide glimpse of provisioned account and any "apps" they provisioned in set up flow.
  * ~~[ ] Create the apps/databases but do not provision. Upon verification of a user, provision any apps/dbs they have not provisioned.~~ (already happens as of https://github.com/aptible/diesel.aptible.com/commit/2d1f2aefeaa7d512b379eb4a6f9f4ce0dbb678d6)
  * [x] verify that deploys check for a verified user.
     * Unverified users are prevented from creating operations in https://github.com/aptible/api.aptible.com/pull/168
  * [x] unverified users see a verify page (needs design for apps/dbs they have created)
  * [x] unverified users see a verify message with button to click to resend verification email

### PR DESCRIPTION
**Pull request by bantic**
Unverified users can still use the UI but they see a persistent
'you are not verified, click here to resend verification email' banner.
In addition, the 'create' buttons for apps, dbs and log drains are not show,
and the '/new' pages for each of those three shows a message saying you must be
verified to create.

fixes #128

@mixonic, for you
@sandersonet please take a look at the UI updates (comments on #128) and
let me know if you'd like to see any changes.

### CODE REVIEW

File: app/apps/new/template.hbs [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
-        <form role="form">
-          {{#unless model.isValid}}
-            <div class="alert alert-warning fade in">
-              <a class="close" data-dismiss="alert">×</a>
-              {{#each model.errors.messages as |message|}}
-                <p>{{message}}</p>
-              {{/each}}
-            </div>
-          {{/unless}}
+{{#if session.currentUser.verified}}
```
Comment (mixonic) [inline app/apps/new/template.hbs]: I wonder if we can easily refactor this into `renderTemplate` calls on the route.

**Comment by bantic** [comment]
@mixonic Some churn in the test files due to me adding some new test helpers and updating old syntax when I revisited the files.

Lots of churn to the payment test:
  * added a function to mock successful payment and pulled out explicit `fillIn` and assertions for stripe/CC data from tests that were not testing that part
  * changed the user data to default to unverified
  * updated the tests that included provisioning to set the user as verified

File: app/components/verification-code-reset/template.hbs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  <div class="row">
+    {{#bs-alert alert="warning" as |component|}}
+      {{#bs-alert-dismiss target=component}}
+         <button type="button" class="close" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+      {{/bs-alert-dismiss}}
+
+      <span class='danger'>Your email is not activated.</span>
+
+      You will not be able to deploy or create new apps until you activate your email.
+      <button class='btn btn-default btn-xs' {{action 'resendVerification'}} {{bind-attr disabled="isDisabled"}}>
```
Comment (mixonic) [inline app/components/verification-code-reset/template.hbs]: I think you mean `{{bind-attr disabled=isDisabled}}`, no quotes?

File: app/templates/-nav-header.hbs [hunk 3]
```diff
@@ -10,3 +10,7 @@
     </ul>
   </div>
 </header>
+
+{{#unless session.currentUser.verified}}
+  {{verification-code-reset store=store}}
```
Comment (mixonic) [inline app/templates/-nav-header.hbs]: Maybe we should just inject store onto components?

File: app/components/verification-code-reset/template.hbs (hunk 2 above)
Comment (bantic) [inline app/components/verification-code-reset/template.hbs]: correct, thanks

File: app/templates/-nav-header.hbs (hunk 3 above)
Comment (bantic) [inline app/templates/-nav-header.hbs]: done

File: app/apps/new/template.hbs (hunk 1 above)
Comment (bantic) [inline app/apps/new/template.hbs]: good call, that works well

**Comment by bantic** [comment]
@mixonic addressed your comments in the last two commits. When the tests goes green this is good to go.

**Comment by bantic** [comment]
closes #51

**Pull request merged**